engine/robot.py

```diff
@@ -57,170 +26,76 @@ class Robot:
             output_limits=(None, None)
         )
 
         # TODO: wrap in try/except (error when calling execute_setup_test.py)
         # write in csv
-        if self.is_store:
-            write_phase_to_csv(self.phase)
+        if self.robot_state.is_store:
+            write_phase_to_csv(self.robot_state.phase)
 
     def update_ekf_step(self):
         zone = ENGINEERING_QUAD  # Used for GPS visualization, make this not hard-coded
-        # self.ekf_var.update_step(self.ekf_var.mu, self.ekf_var.sigma, sensor_module.get_measurement(self.init_gps))
-        self.gps_data = (self.gps.get_gps()["long"], self.gps.get_gps()["lat"])
-        self.imu_data = self.imu.get_gps()
+        # self.robot_state.ekf.update_step(self.robot_state.ekf.mu, self.robot_state.ekf.sigma, sensor_module.get_measurement(self.robot_state.init_gps))
+        self.robot_state.gps_data = (self.robot_state.gps.get_gps()["long"], self.robot_state.gps.get_gps()["lat"])
+        self.robot_state.imu_data = self.robot_state.imu.get_gps()
         x, y = get_vincenty_x(
-            self.init_gps, self.gps_data), get_vincenty_y(self.init_gps, self.gps_data)
+            self.robot_state.init_gps, self.robot_state.gps_data), get_vincenty_y(self.robot_state.init_gps, self.robot_state.gps_data)
         heading = math.degrees(math.atan2(
-            self.imu_data["mag"]["y"], self.imu_data["mag"]["x"]))
+            self.robot_state.imu_data["mag"]["y"], self.robot_state.imu_data["mag"]["x"]))
 
         measurements = np.array([[x], [y], [heading]])
 
-        self.ekf_var.update_step(
-            self.ekf_var.mu, self.ekf_var.sigma, measurements)
-        new_x = self.ekf_var.mu[0][0]
-        new_y = self.ekf_var.mu[1][0]
-        new_heading = self.ekf_var.mu[2][0]
+        self.robot_state.ekf.update_step(
+            self.robot_state.ekf.mu, self.robot_state.ekf.sigma, measurements)
+        new_x = self.robot_state.ekf.mu[0][0]
+        new_y = self.robot_state.ekf.mu[1][0]
+        new_heading = self.robot_state.ekf.mu[2][0]
         new_state = np.array([[new_x], [new_y], [new_heading]])
         return new_state
 
     def travel(self, velocity, omega):
         # Moves the robot with both linear and angular velocity
-        self.state = np.round(integrate_odom(
-            self.state, velocity, omega), 3)
+        self.robot_state.state = np.round(integrate_odom(
+            self.robot_state.state, velocity, omega), 3)
         # if it is a simulation,
-        if self.is_sim:
-            self.truthpose = np.append(
-                self.truthpose, np.transpose(self.state), 0)
+        if self.robot_state.is_sim:
+            self.robot_state.truthpose = np.append(
+                self.robot_state.truthpose, np.transpose(self.robot_state.state), 0)
+        else:
+            imu_data = self.robot_state.imu.get_imu()
+            self.robot_state.state[2] = math.degrees(math.atan2(imu_data["mag"][1], imu_data["mag"][0]))
```

Comment:
What does the else branch of this if statement do - I'm not sure I understand

Reply:
same 🤔 checked main and its not there?? think it might've been an incorrectly resolved merge conflict from main. removed!

Reply:
i think they added it so we have the logic / know how to find the heading from the imu, but we might want to move it in the future

Reply:
ok! lets leave this for now then